### DISCUSSION TITLE
Charging Views - Filter by Time of Day

### DISCUSSION THREAD

**Discussion by igeryu** [teslamate-org/teslamate#3358] [Ideas]
When evaluating a new electrical power provider plan that has free power for certain time windows (nights and/or weekends, etc), it would be very helpful to see what the historical costs of charging in those time windows has been.

I'm not sure how "simple" this is to _implement_, but from the UI, it could be simply a start and stop hour of the day.

Potentially impacted dashboards are `Charging Stats` and `Charges`.

Thank you for your time

**Comment by cwanja** [comment]
Curious - does it even matter? Wouldn't your entire charging behavior change to charge during the free window? Thus you want take a holistic look at your charging behavior during a monthly timeframe?

**Reply by igeryu** [reply]
Well, I would consider if my current charging habits warrant going to any plan had different rates for different parts of the day.  Not only are there "free nights/weekends" plans, but there's now an offering from Tesla for unlimited charging from 10 PM to 6 AM, so I wanted to know how much my charging in that specific timeframe costs me.

Said another way, I don't _want_ to change my charging behavior, instead I want to see how that behavior would fair with other plans.
